apps/apps_publisher.py

```diff
@@ -67,13 +71,6 @@ def __publish(self, app: ComposeApps.App, tag: str):
         app_base_url = self._image_base_url + '/' + app.name
         self._app_tagged_url = app_base_url + ':app-' + tag
         # TODO: Consider implementation of the "publish tool" in DockerRegistryClient
-        out = cmd_exe(self._publish_tool, self._app_tagged_url, cwd=app.dir, capture=True)
-        # The publish command produces output like:
-        # = Publishing app...
-        # |-> app:  sha256:fc73321368c7a805b0f697a0a845470bc022b6bdd45a8b34
-        # |-> manifest:  sha256:e15e3824fc21ce13815aecb0490d60b3a32
-        # We need the manifest sha so we can pin it properly in targets.json
-        needle = b'|-> manifest:  sha256:'
-        sha = out[out.find(needle) + len(needle):].strip()
-        app_hashed_uri = app_base_url + '@sha256:' + sha.decode()
-        return app_hashed_uri
+        cmd_exe(self._publish_tool, '-d', '/tmp/sha', self._app_tagged_url, cwd=app.dir)
+        with open('/tmp/sha') as f:
+            return app_base_url + '@' + f.read().strip()
```

Comment:
As far as I understand this https://github.com/foundriesio/compose-publish/commit/ff9495db64e42a5523f020f21143755721b28c19 dumps a digest in the format we expect here, i.e. `sha256:<hash>`?